File: devops-scm-provider/devops-scm-provider-git/devops-scm-provider-tgit/src/main/java/com/tencent/devops/scm/provider/git/tgit/auth/TGitAuthProviderAdapter.java
package com.tencent.devops.scm.provider.git.tgit.auth;

import com.tencent.devops.scm.api.pojo.auth.IScmAuth;
import com.tencent.devops.scm.sdk.tgit.auth.TGitAuthProvider;

/**
 * tgit 授权提供者适配器
 */
public interface TGitAuthProviderAdapter {

    boolean support(IScmAuth auth);

    TGitAuthProvider get(IScmAuth auth);
}

File: devops-scm-provider/devops-scm-provider-git/devops-scm-provider-tgit/src/main/kotlin/com/tencent/devops/scm/provider/git/tgit/auth/TGitAuthProviderAdapter.kt
package com.tencent.devops.scm.provider.git.tgit.auth

import com.tencent.devops.scm.api.pojo.auth.IScmAuth
import com.tencent.devops.scm.sdk.tgit.auth.TGitAuthProvider

/**
 * TGit授权提供者适配器接口
 */
interface TGitAuthProviderAdapter {
    /**
     * 判断是否支持该授权类型
     * @param auth 授权信息
     * @return 是否支持
     */
    fun support(auth: IScmAuth): Boolean

    /**
     * 获取对应的授权提供者
     * @param auth 授权信息
     * @return 授权提供者实例
     */
    fun get(auth: IScmAuth): TGitAuthProvider
}
